src/socket/mod.rs

```diff
@@ -16,6 +16,7 @@ mod raw;
 mod udp;
 mod tcp;
 mod set;
+mod socket_ref;
```

Comment:
`mod ref`, `use self::ref::Ref` and then reexport as a prefixed name; that's how everything else is done here.

Reply:
`ref` is a keyword, so I can't use `mod ref;`, I'll rename `socket_ref::SocketRef` to `socket_ref::Ref`.